lib/mocha/mockery.rb

```diff
@@ -105,9 +105,12 @@ def state_machines
 
     def mocha_inspect
       message = ''
-      message << "unsatisfied expectations:\n- #{unsatisfied_expectations.map(&:mocha_inspect).join("\n- ")}\n" unless unsatisfied_expectations.empty?
-      message << "satisfied expectations:\n- #{satisfied_expectations.map(&:mocha_inspect).join("\n- ")}\n" unless satisfied_expectations.empty?
-      message << "states:\n- #{state_machines.map(&:mocha_inspect).join("\n- ")}\n" unless state_machines.empty?
+      [
+        ['unsatisfied expectations', unsatisfied_expectations], ['satisfied expectations', satisfied_expectations],
+        ['states', state_machines]
+      ].each do |label, list|
+        message << "#{label}:\n- #{list.map(&:mocha_inspect).join("\n- ")}\n" unless list.empty?
+      end
```

Comment:
While removing the duplication here is good, I'm not convinced that the resultant code is very easy to follow.